### PR TITLE
Fix ENR decode error false positive

### PR DESCRIPTION
**Pull request by dapplion**
**Motivation**

- https://github.com/ChainSafe/lodestar/pull/5034 logs a decode error for valid ENRs

**Description**

- Fix ENR decode error false positive

### CODE REVIEW

File: packages/beacon-node/src/network/discv5/index.ts [hunk 1]
```diff
@@ -73,18 +73,14 @@ export class Discv5Worker extends (EventEmitter as {new (): StrictEventEmitter<E
...
   }
 
   async enr(): Promise<ENR> {
     if (this.status.status === "started") {
-      const enr = this.decodeEnr(await this.status.workerApi.enrBuf());
-      if (enr === undefined) {
-        throw new Error("unable to decode self ENR");
-      }
-      return enr;
+      return ENR.decode(Buffer.from(await this.status.workerApi.enrBuf()));
```
Comment (dapplion) [inline packages/beacon-node/src/network/discv5/index.ts:83]: Our own ENR is not untrusted input, so should throw an error if invalid

**Review by wemeetagain — APPROVED** [review]

File: packages/beacon-node/src/network/discv5/index.ts [hunk 2]
```diff
@@ -123,32 +119,28 @@ export class Discv5Worker extends (EventEmitter as {new (): StrictEventEmitter<E
...
     }
-
-    this.opts.metrics?.discv5.decodeEnrErrorCount.inc(errors);
-    this.opts.metrics?.discv5.decodeEnrAttemptCount.inc(enrBufs.length);
-
     return enrs;
   }
 
-  private decodeEnr(enrBuf: Uint8Array): ENR | undefined {
+  private decodeEnr(enrBuf: Uint8Array): ENR | null {
```
Comment (dapplion) [inline packages/beacon-node/src/network/discv5/index.ts:132]: Dedup code between decodeEnr and decodeEnrs

File: packages/beacon-node/src/network/discv5/index.ts [hunk 3]
```diff
@@ -123,32 +119,28 @@ export class Discv5Worker extends (EventEmitter as {new (): StrictEventEmitter<E
...
       return ENR.decode(Buffer.from(enrBuf));
     } catch (e) {
       this.opts.metrics?.discv5.decodeEnrErrorCount.inc(1);
-    } finally {
-      this.logger.debug("ENR decode error", {enr: Buffer.from(enrBuf).toString("base64url")});
-      this.opts.metrics?.discv5.decodeEnrAttemptCount.inc(1);
+      // Log to recover ENR from logs and debug why it is invalid
+      this.logger.debug("ENR decode error", {
+        enr: Buffer.from(enrBuf).toString("base64url"),
+        error: (e as Error).message,
```
Comment (dapplion) [inline packages/beacon-node/src/network/discv5/index.ts:141]: Add error message to be able to debug, move from finally to catch block

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.6.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.6.0) :tada:

**Pull request merged**